nemo_skills/inference/model/base.py

```diff
@@ -130,6 +149,32 @@ def _maybe_apply_stop_phrase_removal(
         if remove_stop_phrases:
             result["generation"] = trim_after_stop_phrases(result["generation"], stop_phrases)
 
+    def _get_tokenizer(self, tokenizer: str | None) -> Union[ServerTokenizer, WrapperAutoTokenizer, None]:
+        """Get the tokenizer endpoint if available, otherwise initialize from tokenizer string"""
+        tokenizer_endpoint = self._get_tokenizer_endpoint()
```

Comment:
explicit tokenizer should always take precedence. If it's defined we should take it to avoid overriding default. If it's not defined, we can query the endpoint